### PR TITLE
Core/Mod-ZoneScript Update code to current core state

### PR DESCRIPTION
**Pull request by WoWHellgarve-HolyDeeW**
Update mod scripts to current core state

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
thanks for your PR, please update your branch to update also the build configuration

**Comment by Helias** [comment]
build is failing
```
/home/runner/work/mod-pvp-zones/mod-pvp-zones/modules/mod-pvp-zones/src/ZoneScript.cpp:315:9: fatal error: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
        sLog->ShouldLog("[pvp_zones] Debug: active: %i, arena_name: %s, zone_name: %s, last_announcement: %i, last_event: %i, next_announcement: %is", LogLevel::LOG_LEVEL_DEBUG ), config.active, config.current_area_name, config.current_zone_name, config.last_announcement, config.last_event, (config.last_announcement + config.announcement_delay) - GameTime::GetGameTime().count();
        ^~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/mod-pvp-zones/mod-pvp-zones/src/common/Logging/Log.h:128:14: note: expanded from macro 'sLog'
#define sLog Log::instance()
             ^
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/build.make:90: modules/CMakeFiles/modules.dir/mod-pvp-zones/src/ZoneScript.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1199: modules/CMakeFiles/modules.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: src/ZoneScript.cpp [hunk 1]
```diff
@@ -311,7 +312,7 @@ class ZoneCommands : public CommandScript
 
     static bool HandleDebugCommand(ChatHandler* /* handler */)
     {
-        sLog->outString("[pvp_zones] Debug: active: %i, arena_name: %s, zone_name: %s, last_announcement: %i, last_event: %i, next_announcement: %is", config.active, config.current_area_name, config.current_zone_name, config.last_announcement, config.last_event, (config.last_announcement + config.announcement_delay) - sWorld->GetGameTime());
+        sLog->ShouldLog("[pvp_zones] Debug: active: %i, arena_name: %s, zone_name: %s, last_announcement: %i, last_event: %i, next_announcement: %is", LogLevel::LOG_LEVEL_DEBUG ), config.active, config.current_area_name, config.current_zone_name, config.last_announcement, config.last_event, (config.last_announcement + config.announcement_delay) - GameTime::GetGameTime().count();
```
Comment (Winfidonarleyan) [inline src/ZoneScript.cpp:315]: ?

File: src/ZoneScript.cpp [hunk 2]
```diff
@@ -348,21 +349,21 @@ class ZoneWorld : public WorldScript
...
         {
             ZoneLogicScript::EndEvent(handle);
         }
 
         /* announcement stuff */
-        if (config.last_announcement + config.announcement_delay <= sWorld->GetGameTime())
+        if (config.last_announcement + config.announcement_delay <= GameTime::GetGameTime().count())
         {
-            sLog->outString("[pvp_zones] Announcement posted");
+            sLog->ShouldLog("[pvp_zones] Announcement posted", LogLevel::LOG_LEVEL_DEBUG);
```
Comment (Winfidonarleyan) [inline src/ZoneScript.cpp:366]: ?

**Pull request closed without merging**